docker/api/client.py

```diff
@@ -381,19 +386,58 @@ def _stream_raw_result(self, response, chunk_size=1, decode=True):
         for out in response.iter_content(chunk_size, decode):
             yield out
 
-    def _read_from_socket(self, response, stream, tty=False):
+    def _read_from_socket(self, response, stream, tty=False, demux=False):
         socket = self._get_raw_response_socket(response)
 
-        gen = None
-        if tty is False:
+        if tty:
+            # When `tty=True`, there's no framing, we just read the raw data
+            # from the socket. It is as if the streams were multiplexed and
+            # spit out on stdout.
+            gen = ((STDOUT, data) for data in socket_raw_iter(socket))
```

Comment:
Since this is the only place `socket_raw_iter` is used, maybe we can just make that change inside the function instead?

Reply:
done. I got rid of `socket_raw_iter()`, and added a `tty` argument to `iter_frames()`.